mono/mini/interp/interp.c

```diff
@@ -390,8 +390,9 @@ get_virtual_method (InterpMethod *imethod, MonoObject *obj)
 }
 
 static void inline
-stackval_from_data (MonoType *type, stackval *result, char *data, gboolean pinvoke)
+stackval_from_data (MonoType *type_, stackval *result, char *data, gboolean pinvoke)
 {
+	MonoType *type = mini_native_type_replace_type (type_);
```

Comment:
instead of replacing it in the callee, wouldn't it be better to replace it in the caller?